src/hott/09-propositions.rzk.md

```diff
@@ -306,3 +306,179 @@ every fiber `C a`.
+
+#def is-contr-fib-over-image-is-emb
+  ( A B : U)
+  ( f : A → B)
+  ( is-emb-f : is-emb A B f)
+  ( y : A)
+  : is-contr (fib A B f (f y))
+  := is-contr-image-endpoint-based-paths-is-emb A B f is-emb-f y
+
+#def is-contr-is-inhabited-fib-is-emb
+  ( A B : U)
+  ( f : A → B)
+  ( is-emb-f : is-emb A B f)
+  : ( b : B) → is-contr-is-inhabited (fib A B f b)
+  :=
+  ind-fib A B f
+  ( \ b p → is-contr (fib A B f b))
+  ( \ a → is-contr-fib-over-image-is-emb A B f is-emb-f a)
+
+#def is-prop-fib-is-emb
+  ( A B : U)
+  ( f : A → B)
+  ( is-emb-f : is-emb A B f)
+  ( b : B)
+  : is-prop (fib A B f b)
+  :=
+  is-prop-is-contr-is-inhabited (fib A B f b)
+  ( is-contr-is-inhabited-fib-is-emb A B f is-emb-f b)
+
+#def is-contr-image-based-paths-is-contr-is-inhabited-fib
+  ( A B : U)
+  ( f : A → B)
+  ( is-contr-is-inhabited-fib-f : (b : B) → is-contr-is-inhabited (fib A B f b))
+  ( x : A)
+  : is-contr (Σ (y : A) , f x = f y)
+  :=
+  is-contr-equiv-is-contr'
+  ( Σ (y : A) , f x = f y)
+  ( Σ (y : A) , f y = f x)
+  ( total-equiv-family-of-equiv A (\ y → f x = f y) (\ y → f y = f x)
+    ( \ y → equiv-rev B (f x) (f y)))
+  ( is-contr-is-inhabited-fib-f (f x) ((x, refl)))
+
+#def is-emb-is-contr-is-inhabited-fib
+  ( A B : U)
+  ( f : A → B)
+  ( is-contr-is-inhabited-fib-f : (b : B) → is-contr-is-inhabited (fib A B f b))
+  : is-emb A B f
+  :=
+  \ x y →
+  are-equiv-from-paths-is-contr-total A x (\ z → f x = f z)(\ z → ap A B x z f)
+  ( is-contr-image-based-paths-is-contr-is-inhabited-fib A B f is-contr-is-inhabited-fib-f x) y
+
+#def is-emb-is-prop-fib
+  ( A B : U)
+  ( f : A → B)
+  ( is-prop-fib-f : (b : B) → is-prop (fib A B f b))
+  : is-emb A B f
+  :=
+  is-emb-is-contr-is-inhabited-fib A B f
+  ( \ b → is-contr-is-inhabited-is-prop (fib A B f b) (is-prop-fib-f b))
+
+#def is-emb-iff-is-prop-fib
+  ( A B : U)
+  ( f : A → B)
+  : iff (is-emb A B f) ((b : B) → is-prop (fib A B f b))
+  := (is-prop-fib-is-emb A B f, is-emb-is-prop-fib A B f)
+```
+
+## Subtypes
+
+A family of propositions `#!rzk P : A → U` over a type `#!rzk A` may be thought
+of as a predicate.
+
+```rzk
+#def is-predicate
+  ( A : U)
+  ( P : A → U)
+  : U
+  := (a : A) → is-prop (P a)
+```
+
+When `#!rzk P` is a predicate on `#!rzk A` then `#!rzk total-type A P` is
+referred to a subtype of `#!rzk A`.
+
+```rzk
+#def is-emb-subtype-projection
+  ( A : U)
+  ( P : A → U)
+  ( is-predicate-P : is-predicate A P)
+  : is-emb (total-type A P) A (projection-total-type A P)
+  :=
+  is-emb-is-prop-fib (total-type A P) A (projection-total-type A P)
+  ( \ a →
+    is-prop-Equiv-is-prop'
+    ( P a) ( fib (total-type A P) A (projection-total-type A P) a)
+    ( equiv-homotopy-fiber-strict-fiber A P a) (is-predicate-P a))
+```
+
+The subtype projection embedding reflects identifications.
+
+```rzk
+#def subtype-eq-reflection
+  ( A : U)
+  ( P : A → U)
+  ( is-predicate-P : is-predicate A P)
+  ( (a, p) (b, q) : total-type A P)
```

Comment:
It appears that two patterns are not allowed by Rzk, for now the suggestion is to introduce the two pairs separately:

```suggestion
  ( (a, p) : total-type A P)
  ( (b, q) : total-type A P)
```

Reply:
Thanks! I couldn't figure out what rzk was trying to tell me.

I prefer using the patterns here because I think it makes the statement clearest. The latest commit implements your suggestion.